app/models/concerns/profile_fields.rb

```diff
@@ -38,6 +38,24 @@ def canonical?
       parent_profile_id.blank?
     end
 
+    def rules_and_rule_groups
+      custom_arrange(conflicting_relationships(rules, rule_groups), required_relationships(rules, rule_groups))
```

Comment:
If there is no recursion, this is just a rename and the body of custom_arrange can be moved here.